app/graphql/hooks.ts

```diff
@@ -39,19 +39,31 @@ export const useDataCubesMetadataQuery = (options: {
 
     const run = async () => {
       const queries = await Promise.all(
-        cubeFilters.map((cubeFilter) =>
-          client
+        cubeFilters.map((cubeFilter) => {
+          const cubeVariables = { locale, sourceType, sourceUrl, cubeFilter };
+          const cached = client.readQuery<
+            DataCubeMetadataQuery,
+            DataCubeMetadataQueryVariables
+          >(DataCubeMetadataDocument, cubeVariables);
+
+          if (cached) {
+            return Promise.resolve(cached);
+          }
+
+          if (!result.fetching) {
+            setResult({
+              ...result,
+              fetching: true,
+            });
+          }
```

Comment:
I am wondering why we need a setResult inside each promise ? I think it would be only necessary to have one outside the map ?

Why do we need to read explicitly from the cache ? I thought that client.query would do that for us ?

Reply:
Ah is it so that we do not have a flash of "loading" when all the queries are already in the cache ? I guess the urql way here would be to pipe the client.query result and subscribe to some kind of "request" event from it, setting the fetching state there. I think this "request" event would not be fired when the query would come from the cache.

Reply:
It was done to properly set the `fetching` property – it we read all queries from cache, I think we shouldn't set fetching to `true`, to avoid flashes of loading indicators – and when at least one query is not read from cache, then we set the fetching directly from the promise, as otherwise we would have awaited them all and not set the `fetching = true` at all.

Does it make sense? 😅 Basically instead of always having `setFetching(true)` at the top, I wanted to only set this when we actually didn't hit the cache 👀

Reply:
That said, maybe the urql way is not very understandable either, I have trouble exactly making sense of it in their source. https://github.com/urql-graphql/urql/blob/main/packages/react-urql/src/hooks/useQuery.ts#L346

Reply:
And yes, it would also be read from cache when using `client.query`, but then it was easier to do `readQuery` so we know when to set the `fetching` prop